client/components/marketing-survey/cancel-jetpack-form/index.tsx

```diff
@@ -65,6 +71,35 @@ const CancelJetpackForm: React.FC< Props > = ( {
 		isSiteAutomatedTransfer( state, purchase.siteId )
 	);
 
+	const cancellationOffer = useSelector( ( state ) => {
+		const offers = getCancellationOffers( state, purchase.id );
+		if ( offers.length > 0 ) {
+			return offers[ 0 ];
+		}
+
+		return null;
+	} );
+
+	const fetchingCancellationOffers = useSelector( ( state ) =>
+		isFetchingCancellationOffers( state )
+	);
+
+	const isOfferPriceSameOrLowerThanPurchasePrice = useMemo( () => {
+		return cancellationOffer ? purchase.amount >= cancellationOffer.originalPrice : false;
+	}, [ cancellationOffer, purchase ] );
+
+	const offerDiscountBasedFromPurchasePrice = useMemo( () => {
+		// toFixed will round to the 3rd decimal place here.
+		// This will catch floats like 19.9999 and round them to the closest integer.
+		return cancellationOffer
+			? Math.floor(
+					Number.parseFloat(
+						Number( 1 - cancellationOffer.rawPrice / purchase.amount ).toFixed( 3 )
+					) * 100
+			  )
+			: 0;
```

Comment:
Besides/Instead commentary above, we could consider breaking down this complex calculation into several consecutive variables describing what is happening on every level of the calculation

Reply:
I like this idea, should make it easier to digest.